src/amo/components/UserProfileEditNotifications/index.js

```diff
@@ -0,0 +1,120 @@
+/* @flow */
+import makeClassName from 'classnames';
+import * as React from 'react';
+import { compose } from 'redux';
+import invariant from 'invariant';
+
+import translate from 'core/i18n/translate';
+import LoadingText from 'ui/components/LoadingText';
+import type { UserType } from 'amo/reducers/users';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+
+// We disable the eslint rule below because Flow complains about unreachable
+// code if we add a `return` statement in the `default` case.
+// eslint-disable-next-line consistent-return
+export const getLabelText = (i18n: I18nType, name: string): string => {
+  switch (name) {
+    case 'announcements':
+      return i18n.gettext(`stay up-to-date with news and events relevant to
+        add-on developers (including the about:addons newsletter)`);
+    case 'individual_contact':
+      return i18n.gettext(`Mozilla needs to contact me about my individual
+        add-on`);
+    case 'new_features':
+      return i18n.gettext('new add-ons or Firefox features are available');
+    case 'new_review':
+      return i18n.gettext('someone writes a review of my add-on');
+    case 'reply':
+      return i18n.gettext('an add-on developer replies to my review');
+    case 'reviewer_reviewed':
+      return i18n.gettext('my add-on is reviewed by a reviewer');
+    case 'sdk_upgrade_fail':
+      return i18n.gettext('my sdk-based add-on cannot be upgraded');
+    case 'sdk_upgrade_success':
+      return i18n.gettext('my sdk-based add-on is upgraded successfully');
+    case 'upgrade_fail':
+      return i18n.gettext(`my add-on's compatibility cannot be upgraded`);
+    case 'upgrade_success':
+      return i18n.gettext(`my add-on's compatibility is upgraded successfully`);
+    default:
+      invariant(false, `No corresponding label for notification "${name}"`);
```

Comment:
Not sure what to do here. I saw this in the react code, and I find it quite elegant but then we have a coverage miss.

Reply:
Here's an idea: change `createUserNotificationsResponse` to accept an array which can override the array which is returned by default. Then write a test where you create an array of notifications with a "bad" name (it can be the only notification in the array), and pass it into `createUserNotificationsResponse` and use the resulting array of notifications in a call to `loadUserNotifications`. When you render the component that should trigger the error and give us coverage for this line.

In fact it may be overkill to do all of that with `createUserNotificationsResponse` because all you're doing is passing in an array and expecting the same array back, but I suppose if something were to change in the future about the response then it would be nice to have it encapsulated there.

Reply:
@bobsilverberg thanks for the review! You're right, I could get coverage for this with unknown notifications, but what should be the behavior here? `invariant` only? Throwing an exception?

Same question for the UI: if we have no label, then what? We don't display anything for that notification?

That's why I kept things simple, because `invariant` will trigger a warning for us (developers) and I believe not returning anything will just not display a label for that notification (and that won't break the UI entirely).

Reply:
Well, in the test I would expect `invariant` to throw an exception, which we would assert (although we don't usually include assertions for calls to `invariant`, I think this is an exception (pun inteneded!)).

In terms of the display, I guess if we have no label we probably shouldn't be showing the notification input at all. It's bad not to show an input when there's a field for it, but I don't see any point in showing an input without a label. So I think the code for `createNotification` should return nothing if there is no label. Do you agree?

Reply:
I do agree. Then, next question is: I can maybe return `null` in the `default` case and log a warning in `createNotification` to let developers know why one notification is missing. WDYT? That would be testable as usual and we would not have to introduce a test case to check `invariant`.

Reply:
Sure, that sounds good.